backend/app/services/room.test.js

```diff
@@ -0,0 +1,110 @@
+import nock from "nock";
+import fs from "fs";
+import path from "path";
+import fetchRooms from "./rooms";
+
+
+describe(".fetchRooms", () => {
+  const validRoomsData = [{ id: "afce3adc-00f0-4424-923c-d10cd72939b3", name: "Logos" }];
+  describe("when strategy is ENVIRONMENT", () => {
+    const strategy = "ENVIRONMENT";
+    it("should load from env ROOMS_DATA", (done) => {
+      process.env.ROOMS_DATA = JSON.stringify(validRoomsData);
+      fetchRooms(strategy).then((rooms) => {
+        expect(rooms).toEqual(validRoomsData);
+      }).then(done);
+    });
+
+    it("should reject the promise when invalid JSON defined", (done) => {
+      process.env.ROOMS_DATA = "INVALID_JSON";
+      fetchRooms(strategy).then((rooms) => {
+        done.fail(`Need fail on parse invalid json: ${rooms}`);
+      }).catch((error) => {
+        expect(error.message).toEqual("Unexpected token I in JSON at position 0");
+        done();
+      });
+    });
+  });
+
+  describe("when strategy is REMOTE", () => {
+    const strategy = "REMOTE";
+    const myUrlData = "http://minhaurl.com";
+    beforeEach(() => {
+      process.env.ROOMS_DATA = myUrlData;
+    });
+
+    it("should load from a URL", (done) => {
+      nock(myUrlData)
+        .get("/")
+        .reply(200, validRoomsData);
+
+      fetchRooms(strategy).then((rooms) => {
+        expect(rooms).toEqual(validRoomsData);
+      }).then(done);
+    });
+
+    it("should reject the promise when invalid JSON defined", (done) => {
+      nock(myUrlData)
+        .get("/")
+        .reply(200, "INVALID_JSON");
```

Comment:
Se tem erro, não deveria retornar 200, deveria retornar 400 (Bad Request) ou um outro status code de erro.

Reply:
É que nesse caso, o request não foi deu errado, mas o JSON que a pessoa resolveu definir estava num formato inválido. Então a request retorna sucesso, mas é um conteudo que não tem como fazer um JSON.parse.  Faz sentido?

Reply:
Mas não conseguir fazer o `JSON.parse` não é um problema? Você não está retornando um erro? Então faz sentido um status code de sucesso, entendeu?
Como o problema está na configuração do server, o status code pode ser 500 (Internal Server Error), pq o sistema não era nem para estar rodando com a lista de salas quebrada.

Reply:
Mas o  JSON.parse não funcionar é um problema na Matrix, não no request do cara. Eu dei um GET e ele me devolveu algo, se era o que eu queria ou não é outra história. Se eu uso um status code que não é sucesso, é como se o servidor soubesse que tá errado, mas aqui eu quero simular o caso do cara definir um JSON errado e não se ligar (EX: o cara copia e cola uma nova sala e esquece de remover a , no final). 
Faz sentido? Qualquer coisa a gente faz uma call, talvez seja mais fácil de explicar

Reply:
I was looking for some best practices about this discussion and I find [this](https://blog.pedrofelix.org/2016/10/10/health-check-apis-500-status-codes-and-media-types/) article. The article explains what the @megatroom meant.  

>> Apparently, it is also a common practice for a GET request to these resources to return a 500 status code if any of the internal components reports a problem. For instance, the Dropwizard documentation states

Reply:
Pra mim ainda não faz sentido esse curso de ação. Mas eu alterei o teste para retornar erro nesse caso.